src/components/LeadsTable/leadsSlice.ts

```diff
@@ -0,0 +1,119 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+
+interface Lead {
+   _id: string;
+   name: string;
+   email: string;
+   consentsAccepted: boolean;
+   createdAt: string;
+   updatedAt: string;
+   __v: number;
+}
+
+interface LeadsState {
+   leads: Lead[];
+   isLoading: boolean;
+   error: string | null;
+   sortKey: 'name' | 'email' | 'date' | 'consentsAccepted';
+   sortOrder: 'asc' | 'desc';
+}
+
+const initialState: LeadsState = {
+   leads: [],
+   isLoading: false,
+   error: null,
+   sortKey: 'date',
+   sortOrder: 'desc',
+};
+
+export const fetchLeads = createAsyncThunk('leads/fetchLeads', async () => {
+   const response = await fetch('https://training.nerdbord.io/api/v1/leads');
+   const data = await response.json();
+   return data;
+});
+
+export const sortLeadsByColumn = createAsyncThunk(
+   'leads/sortLeadsByColumn',
+   async (column: 'name' | 'email' | 'date' | 'consentsAccepted', { getState, dispatch }) => {
+      const { leads, sortKey, sortOrder } = (getState() as { leads: LeadsState }).leads;
+      let sortedLeads = [...leads];
+
+      if (column === sortKey) {
+         sortedLeads.reverse();
```

Comment:
**issue:** We need to kill this `ifology` here. Redux slices should be as pure, and simple as possible. Doing it this way is an antipattern. It should be moved to UI layer